space-plugins/nuxt-base/server/utils/getAppSession.ts

```diff
@@ -29,93 +24,25 @@ type AppSessionQuery = {
 // (11) https://<USER-DOMAIN>/api/connect/callback?code=xxxx&state=xxxx&space_id=xxxx
 
 // Finished. Authenticated successfully.
-// (12) https://<USER-DOMAIN>/?spaceId=xxxx&userId=xxxx
-
-// If a user visits this plugin with the `sb.auth` token already set, they will be redirected to ↓
-// (13) https://<USER-DOMAIN>/?space_id=xxxx&user_id=xxxx
-// instead of spaceId and userId.
-// It should be more consistent, but at least the `nuxt-base` layer takes care of it now.
+// (12) https://<USER-DOMAIN>/?space_id=xxxx&user_id=xxxx
 
 export const getAppSession = async (event: H3Event) => {
-	const appSessionQuery = extractAppSessionQuery(event);
-	if (!isAppSessionQuery(appSessionQuery)) {
-		return;
-	}
-
 	const appConfig = useAppConfig();
-	const sessionStore = sessionCookieStore(getAuthHandlerParams(appConfig.auth))(
-		{
-			req: event.node.req,
-			res: event.node.res,
-		},
-	);
-
-	return await sessionStore.get(appSessionQuery);
-};
-
-function extractAppSessionQuery(event: H3Event): AppSessionQuery | undefined {
+	const sessionStore = getSessionStore(getAuthHandlerParams(appConfig.auth))({
+		req: event.node.req,
+		res: event.node.res,
+	});
 	const appSession = event.context.appSession;
-	const query = getQuery(event);
-
-	if (appSession?.spaceId && appSession?.userId) {
-		// When a page is already authenticated on the server side,
-		// and it's rendering a page that includes `useFetch('...', { server: true })`,
-		// Nuxt directly runs the serverless function on the server side
-		// (without an actual HTTP request).
-		//
-		// In that case `event.context.appSession` exists.
-		return convertToAppSessionQuery(appSession?.spaceId, appSession?.userId);
+	if (appSession && appSession.spaceId && appSession.userId) {
```

Comment:
maybe we can even expose isAppSession validation method which we could use here?

Reply:
that's a good point.